### PR TITLE
:wrench: fix the search bar & add patient fix #185

### LINKED ISSUE #185: Add the possibility to search by patient in list of prescription

### PR DESCRIPTION
**Pull request by issa-diallo**
Add the possibility to search by patient in list of prescription

### CODE REVIEW

File: src/components/Prescriptions.tsx [hunk 1]
```diff
@@ -42,35 +53,36 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
...
-
-        patientResults.forEach(({ id, patient }) => {
-          patientsData[id] = patient
+        setPatients((prevPatients) => {
+          const newPatientsData = { ...prevPatients }
+          patientResults.forEach(({ id, patient }) => {
+            newPatientsData[id] = patient
+          })
+          return newPatientsData
         })
```
Comment (AndreMiras) [inline src/components/Prescriptions.tsx:62]: Functional programming for the win, how about:
```ts
setPatients(prevPatients => {
    return patientResults.reduce((newPatientsData, { id, patient }) => {
      return { ...newPatientsData, [id]: patient }
    }, { ...prevPatients })
})
```

File: src/components/Prescriptions.tsx [hunk 2]
```diff
@@ -42,35 +53,36 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
...
+        )
+        setFilteredPrescriptions(filteredData)
       } catch (error) {
         console.error(error)
       }
     }
 
     // eslint-disable-next-line no-void
     void fetchPatients()
+  // eslint-disable-next-line
```
Comment (AndreMiras) [inline src/components/Prescriptions.tsx:75]: Be more explicit on what we're ignoring

File: src/components/Prescriptions.tsx [hunk 3]
```diff
@@ -24,9 +24,20 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
...
+
+  const filterByPrescriptions = useCallback((prescription: Prescription, searchValue: string): boolean => {
+    const doctorMatch = prescription.prescribing_doctor.toLowerCase().includes(searchValue.toLowerCase())
+    const patient = patients[prescription.id]
+    assert(patient)
+    const patientFullName = `${patient.firstname} ${patient.lastname}`.toLowerCase()
+    const patientMatch = patientFullName.includes(searchValue.toLowerCase())
+
+    return doctorMatch || patientMatch
+  }, [patients])
```
Comment (AndreMiras) [inline src/components/Prescriptions.tsx:40]: We could break this down a bit to give it a bit more semantic

File: src/components/Prescriptions.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/components/Prescriptions.tsx:62]: I've create a function

File: src/components/Prescriptions.tsx (hunk 2 above)
Comment (issa-diallo) [reply src/components/Prescriptions.tsx:75]: I add react-hooks/exhaustive-deps

File: src/components/Prescriptions.tsx (hunk 3 above)
Comment (issa-diallo) [reply src/components/Prescriptions.tsx:40]: I created many functions

File: src/components/Prescriptions.tsx [hunk 4]
```diff
@@ -24,9 +24,44 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
...
 
   const [filteredPrescriptions, setFilteredPrescriptions] = useState<Prescription[]>(prescriptions)
+  const [searchPatient, setSearchPatient] = useState<string>('')
 
   // Using Record to get a key-value array
-  const [patients, setPatients] = useState<Record<number, Patient | null>>({})
+  const [patients, setPatients] = useState<Record<number, Patient>>({})
+
+  // Function to check if the prescription's doctor name matches the search
+  const doesPrescriptionDoctorMatch = (prescription: Prescription, searchValue: string): boolean => {
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/Prescriptions.tsx]: fixed

[the remark this answers is not in the document]
Comment (AndreMiras) [reply src/components/Prescriptions.tsx]: isDoctorMatchING*

File: package.json [hunk 5]
```diff
@@ -72,6 +72,7 @@
   "devDependencies": {
     "@types/lodash": "^4.14.181",
     "@types/react-datepicker": "^4.11.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (AndreMiras) [inline package.json:75]: why do we need this for?

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**